cmake/compiler_flags.cmake

```diff
@@ -23,7 +23,7 @@ else()
             set(warnings "${warnings} -Wno-missing-field-initializers")
         endif()
 
-        set(warnings "${warnings} -Wuseless-cast -Wlogical-op")
```

Comment:
This fails when we generate files with `protoc`, and we don't have control on those files.

I removed the option here, because I learnt that we should ideally even give the `-Werror` as an option to the user, because our code might fail on some compilers because of that. So ideally we would want a cmake option (e.g. `CMAKE_WARNINGS_AS_ERRORS`).

Using `-Wno-useless-cast` in the backend only is not a good solution because it forces users of `libbackend` to inherit it in their code.

Reply:
I agree. We only want warnings as errors during development.